code/game/objects/structures/crates_lockers/closets/secure/vault.dm

```diff
@@ -0,0 +1,52 @@
+/obj/structure/closet/secure_closet/vault
+	name = "vault locker"
+	desc = "For when you absolutely need to keep something safe."
+	icon = 'icons/obj/closet.dmi'
+	icon_state = "vault1"
+	density = 1
+	opened = 0
+	large = 1
+	locked = 1
+	anchored = 1 //immovable
+	icon_closed = "vault"
+	icon_locked = "vault1"
+	icon_opened = "vaultopen"
+	icon_broken = "vaultbroken"
+	icon_off = "vaultoff"
+	wall_mounted = 0 //never solid (You can always pass over it)
+	health = 20000
+
+/obj/structure/closet/secure_closet/vault/ex_act(var/severity) //bomb-proof
+	var/list/bombs = search_contents_for(/obj/item/device/transfer_valve)
+	if(!isemptylist(bombs))
+		..(severity)
```

Comment:
@Intigracy 
Additionally, the code was made like this for the freezer so the locker doesn't withstand explosions triggered by bombs placed inside the locker.
If I changed it to simply remove health, someone could survive a bomb detonated inside the locker by hiding in that same locker with the bomb.
I don't know if there's a good way to check the contents of the locker for a bomb *and* make sure it's the same bomb that caused the explosion that caused the check.